akka-docs/rst/scala/code/docs/stream/ActorSubscriberDocSpec.scala

```diff
@@ -79,11 +82,13 @@ class ActorSubscriberDocSpec extends AkkaSpec {
 
     //#actor-subscriber-usage
     val N = 117
-    Source(1 to N).map(WorkerPool.Msg(_, replyTo))
+    val worker = Source(1 to N).map(WorkerPool.Msg(_, replyTo))
       .runWith(Sink.actorSubscriber(WorkerPool.props))
     //#actor-subscriber-usage
 
+    watch(worker)
     receiveN(N).toSet should be((1 to N).map(WorkerPool.Done).toSet)
```

Comment:
The new test failure is not about too short timeout. In the log you can see
```
[WARN] [08/24/2016 12:48:21.862] [ActorSubscriberDocSpec-akka.actor.default-dispatcher-2] [akka://ActorSubscriberDocSpec/user/StreamSupervisor-46/flow-0-1-actorSubscriberSink] received dead letter from Actor[akka://ActorSubscriberDocSpec/user/StreamSupervisor-46/flow-0-1-actorSubscriberSink/$b#1432547271]: Reply(116)
[WARN] [08/24/2016 12:48:21.863] [ActorSubscriberDocSpec-akka.actor.default-dispatcher-2] [akka://ActorSubscriberDocSpec/user/StreamSupervisor-46/flow-0-1-actorSubscriberSink] received dead letter from Actor[akka://ActorSubscriberDocSpec/user/StreamSupervisor-46/flow-0-1-actorSubscriberSink/$c#2143819031]: Reply(117)
```

i.e. the actor was stopped before the replies were handled.

In `case OnComplete` you can only stop if queue is empty, and in `case Reply` you should stop `if (canceled && queue.isEmpty`